app/lib/dartdoc/backend.dart

```diff
@@ -156,12 +156,16 @@ class DartdocBackend {
     // deleted, but the [removeObsolete] should be able to validate it.
     await deleteFromBucket(_storage, entry.inProgressObjectName);
 
-    await dartdocMemcache?.invalidate(entry.packageName, entry.packageVersion);
+    await Future.wait([
+      cache.dartdocEntry(entry.packageName, entry.packageVersion).purge(),
```

Comment:
The three related purge may be worth to move to the cache class? Similar to the `invalidatePackageCache` method. (Or maybe, we should have only one `invalidatePackageCache` that also purges the non-versioned ones e.g. cache.dartdocEntry(entry.packageName, 'latest') and the cache.dartdocApiSummary(entry.packageName))

Reply:
I would like the cache class to have no auxiliary methods or magic... just be a mechanical mapping from pattern to `Entry<T>`.

It's only the frontend backend where purging is actually called many places..